### PR TITLE
🧼 Clean up Metal implementation

### PR DESCRIPTION
**Pull request by FlannyH**
- Make all the `#[derive()]`s consistent with the Dx12 implementation
- Improve the formatting slightly by adding newlines
- Add debug label for memory blocks
- Remove unused structs

### CODE REVIEW

**Review by Jasper-Bekkers — APPROVED** [review]

**Review by MarijnS95 — COMMENTED** [review]
Other cleanups you could do (that's not visible in the diff):

- Remove functions that always panic via `todo!()`;
- Change `heap_types.iter()` in `Allocator::new()` to `into_iter()`, so that you can remove `heap_descriptor.clone()`;
- Remove `_size` from `MemoryBlock` if it's never read and not public?

File: src/metal/mod.rs [hunk 1]
```diff
@@ -129,21 +131,28 @@ impl<'a> AllocationCreateDesc<'a> {
...
+
 pub struct Allocator {
     device: Arc<metal::Device>,
     debug_settings: AllocatorDebugSettings,
     memory_types: Vec<MemoryType>,
     allocation_sizes: AllocationSizes,
 }
+
+#[derive(Debug)]
 pub struct AllocatorCreateDesc {
```
[the remark this answers is not in the document]
Comment (FlannyH) [reply src/metal/mod.rs:143]: > `ash::Instance` doesn't implement `std::fmt::Debug`
the trait `std::fmt::Debug` is not implemented for `ash::Instance`
I'll leave that for now

File: src/metal/mod.rs [hunk 2]
```diff
@@ -160,6 +169,10 @@ impl MemoryBlock {
         heap_descriptor.set_size(size);
 
         let heap = Arc::new(device.new_heap(heap_descriptor));
+        heap.set_label(&format!(
+            "MemoryBlock {:?}",
+            heap_descriptor.resource_options(),
```
[the remark this answers is not in the document]
Comment (FlannyH) [reply src/metal/mod.rs:173]: Fair enough. Is it work adding another param to this function just for a debug label though?

File: src/metal/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarijnS95) [reply src/metal/mod.rs:143]: That should have been, unless we lost the `"debug"` feature. Okay to leave it off then.

File: src/metal/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarijnS95) [reply src/metal/mod.rs:173]: Maybe this function should calculate the heap descriptor unless there's value in reusing it?

[the remark this answers is not in the document]
Comment (FlannyH) [reply src/metal/mod.rs:173]: Hmm I'm not sure

**Review by MarijnS95 — APPROVED** [review]

**Comment by MarijnS95** [comment]
Really curious why automerge pulled in the PR even before the CI run completed. If failed after all.

**Pull request merged**